docs/index.js

```diff
@@ -0,0 +1,621 @@
+    await rpc.connect();
+
+    try {
+      const P2SHAddress = document.getElementById("sendP2SHAddress").innerText;
+      const sendScriptBuilder = kaspaWasm.ScriptBuilder.fromScript(
+        document.getElementById("sendScript").innerText,
+      );
+
+      let P2SHEntries = [];
+      while (P2SHEntries.length === 0) {
+        const P2SHUTXOs = await rpc.getUtxosByAddresses([
+          P2SHAddress.toString(),
+        ]);
+        P2SHEntries = P2SHUTXOs.entries;
+
+        if (P2SHEntries.length === 0) {
+          await new Promise((resolve) => setTimeout(resolve, 1000));
+        }
+      }
+
+      const P2SHEntry = P2SHEntries[0];
+      const entry = {
+        address: P2SHEntry.address.toString(),
+        amount: kaspaWasm.sompiToKaspaString(P2SHEntry.amount),
+        scriptPublicKey: JSON.parse(P2SHEntry.scriptPublicKey.toString()),
+        blockDaaScore: P2SHEntry.blockDaaScore.toString(),
+        outpoint: JSON.parse(P2SHEntry.outpoint.toString()),
+      };
+      const revealTxId = await kastle.signAndBroadcastTx("testnet-10", [], {
+        priorityEntries: [entry],
+        scripts: [
+          {
+            scriptHex: sendScriptBuilder.toString(),
+            inputIndex: 0,
+          },
+        ],
+        priorityFee: "0.02",
+      });
+      document.getElementById("cancelTradeTxId").innerText = revealTxId;
+      document.getElementById("cancelErrorKRC20").innerText = "";
+    } catch (error) {
+      document.getElementById("cancelErrorKRC20").innerText = error.message;
+    } finally {
+      rpc.disconnect();
+    }
+  });
+
+document.getElementById("krcBuyReveal").addEventListener("click", async () => {
+  const txJson = document.getElementById("sendPreparedTx").innerText;
+  const tx = kaspaWasm.Transaction.deserializeFromSafeJSON(txJson);
+
+  const address = document.getElementById("address").innerText;
+  const rpc = new kaspaWasm.RpcClient({
+    url: "wss://ws.tn10.kaspa.forbole.com/borsh",
+    networkId: "testnet-10",
+  });
+  await rpc.connect();
+
+  try {
+    const amount = tx.outputs
+      .map((output) => output.value)
+      .reduce((a, b) => a + b, 0n);
+    const entries = [];
+    const utxos = await rpc.getUtxosByAddresses([address]);
+    let total = 0n;
+    const fee = kaspaWasm.kaspaToSompi("0.01");
+    while (total < amount + fee) {
+      const entry = utxos.entries.pop();
+      entries.push(entry);
+      total += entry.amount;
+    }
+
+    const newInputs = entries.map((entry) => {
+      return {
+        previousOutpoint: entry.outpoint,
+        sequence: 0n,
+        sigOpCount: 1,
+        utxo: entry,
+      };
+    });
+
+    tx.inputs = [...tx.inputs, ...newInputs];
+    const balanceOfInputs = tx.inputs
+      .map((input) => input.utxo.amount)
+      .reduce((a, b) => a + b, 0n);
+    const change = balanceOfInputs - amount - fee;
+    tx.outputs = [
+      ...tx.outputs,
+      { value: change, scriptPublicKey: kaspaWasm.payToAddressScript(address) },
+    ];
```

Comment:
Buyer or Market can add new inputs and outputs to the signed transaction by seller to complete the KRC20 trade